packages/app/src/systems/Send/machines/sendMachine.ts

```diff
@@ -0,0 +1,298 @@
+        on: {
+          BACK: {
+            actions: ['goToHome'],
+          },
+        },
+        states: {
+          checking: {
+            always: [{ target: 'delay', cond: 'canCalcFee' }],
+          },
+          delay: {
+            after: {
+              600: 'creatingTx',
+            },
+          },
+          creatingTx: {
+            invoke: {
+              src: 'createTx',
+              data: (ctx: MachineContext) => ({ input: ctx.inputs }),
+              onDone: [
+                {
+                  target: 'failed',
+                  actions: ['assignTxRequestError'],
+                  cond: FetchMachine.hasError,
+                },
+                {
+                  actions: ['assignTxRequest'],
+                  target: 'waitingConfirm',
+                },
+              ],
+            },
+          },
+          waitingConfirm: {
+            on: {
+              CONFIRM: {
+                target: '#(machine).confirming',
+              },
+            },
+          },
+          failed: {
+            type: 'final',
+          },
+        },
+      },
+      confirming: {
+        initial: 'idle',
+        on: {
+          BACK: {
+            target: 'idle.waitingConfirm',
+          },
+        },
+        states: {
+          idle: {
+            on: {
+              CONFIRM: {
+                target: 'sendingTx',
+              },
+            },
+          },
+          sendingTx: {
+            invoke: {
+              src: 'sendTx',
+              data: (ctx: MachineContext) => ({ input: ctx }),
+              onDone: [
+                {
+                  target: 'idle',
+                  actions: ['assignTxApproveError'],
+                  cond: FetchMachine.hasError,
+                },
+                {
+                  actions: ['assignTxApprove'],
+                  target: 'success',
+                },
+              ],
+            },
+          },
+          success: {
+            type: 'final',
+            entry: ['showSuccessMessage', 'goToHome'],
+          },
+        },
+      },
+    },
+    on: {
+      SET_ASSET: {
+        actions: ['assignAsset'],
+        target: 'idle',
+      },
+      SET_ADDRESS: {
+        actions: ['assignAddress'],
+        target: 'idle',
+      },
+      SET_AMOUNT: {
+        actions: ['assignAmount'],
+        target: 'idle',
+      },
+      RESET: {
+        actions: ['reset'],
+        target: 'idle',
+      },
+    },
+  },
+  {
+    actions: {
+      reset: resetContext(),
+      assignAsset: assignWithInput('inputs.asset'),
+      assignAddress: assignWithInput('inputs.address'),
+      assignAmount: assignWithInput('inputs.amount'),
+      assignWallet: assignWithData('inputs.wallet'),
+      assignTxRequest: assignWithData('response', true),
+      assignTxApprove: assignWithData('response', true),
+      assignTxRequestError: assignErrors('txRequestError'),
+      assignTxApproveError: assignErrors('txApproveError'),
```

Comment:
This looks like a great abstraction, but we should avoid including it in this PR as it will create a new pattern only used here. Looking at the feature, this kind of thing can create many machines without the same patterns. A better approach is to split the PR into phases.

1. Add the new abstraction
2. Refactor all the machines to use the new abstraction
3. Use the new abstraction here, as all the code base is updated.

Reply:
Can I let this PR with this modifications and then open a new one including them in the other machines @luizstacio ?

Reply:
In my opinion, we should focus on refactoring it first and after add new abstractions. When doing abstractions we create new patterns to implement things that make current code that can be done or in progress already legacy. A good approach for this kind of thing would be to create an issue and do when is a good time. This would solve the two things as all the code base will follow the pattern, and after all, the code would be refactored.

But I want another opinion from the team, not to be the single arbiter here. @LuizAsFight

Reply:
I agree. We should not merge this part because:
- it's orthogonal to the main objective of the issue: Create send transaction screen
- we should not just implement a whole new pattern when we already have ~10 machines created in other way


Could you open one issue and elaborate what will be the benefits of refactor to include a new layer to manage actions, with a bunch of helpers and complex object management? then we can continue talking from there